src/components/ProductForList.js

```diff
@@ -0,0 +1,95 @@
+import { compareAsc, sub } from 'date-fns';
+import {
+  getDoc,
+  updateDoc,
+  arrayUnion,
+  arrayRemove,
+  doc,
+} from 'firebase/firestore';
+import { db } from '../lib/firebase';
+import { useState } from 'react';
+import { Link } from 'react-router-dom';
+import { getListFromDB } from '../lib/api';
+
+export const ProductForList = ({ item, handleDeleteAttempt, token }) => {
+  const getDateDiff = () => {
+    if (!item.lastPurch) return;
+
+    //Dates
+    const currentTime = new Date();
+    const purchaseDate = item.lastPurch.toDate();
+    const oneDayAgo = sub(currentTime, {
+      days: 1,
+    });
+
+    const result = compareAsc(purchaseDate, oneDayAgo);
+    if (result === -1) {
+      return false;
+    }
+    return true;
+  };
+
+  const [isBought, setIsBought] = useState(getDateDiff());
+
+  const updatePurchaseTimeDB = async () => {
+    //Encuentra la lista.
+    const listCollection = getListFromDB(token);
+    const list = await getDoc(listCollection);
+
+    if (list.exists()) {
+      const itemsFromList = list.data().items;
+
+      //Encuentra el item a actualizar.
+      const thisItem = itemsFromList.find((itemToCheck) => {
+        return itemToCheck.name === item.name;
+      });
+
+      const listRef = doc(db, 'lists', token);
+      const thisItemUpdated = { ...thisItem, lastPurch: new Date() };
+      await updateDoc(listRef, { items: arrayUnion(thisItemUpdated) });
+      await updateDoc(listRef, { items: arrayRemove(thisItem) });
+    }
+  };
+
+  const handleCheck = async () => {
+    //Si se desmarca, no actualiza la DB.
+    if (isBought) {
```

Comment:
I found this function hard to read for some reason. I'm thinking that it could be better organized so that the `return` is not necessary:
```suggestion
   setIsBought(!isBought);
   if(!isBought) {
       updatePurchaseTimeDB();
   }
```

Reply:
Absolutely! Le sobraban varias líneas. :)